### PR TITLE
Update internal tool install instructions to include pyenv, editable, and force

### PR DESCRIPTION
**Pull request by bnchrch**
## Overview
Updates the install instructions for internal tools to
1. Ensure that pyenv is used
2. Pipx is reinstalled on install
3. Pipx tools are installed in editable mode to ensure auto update

### CODE REVIEW

File: airbyte-ci/connectors/pipelines/README.md [hunk 1]
```diff
@@ -22,17 +22,33 @@ This project requires Python 3.10 and pipx.
...
 ```bash
-pipx install airbyte-ci/connectors/pipelines/ --force
+# ensure you are in the correct directory
+cd airbyte-ci/connectors/pipelines/
+
+# ensure you have the correct python version
+pyenv install
+
+# install airbyte-ci
+pipx install --editable --force .
```
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/README.md:51]: We could take this one step further with a shell script we can call like so
```
./install-internal-tool airbyte-ci/connectors/pipelines/
```

e.g.
```sh
#!/bin/bash

# Change to the given directory
cd "$1" || { echo "Directory does not exist"; exit 1; }

# Update brew and install pyenv
brew update
brew install pyenv

# Ensure correct Python version
pyenv install

# Install pipx
python -m pip install --user pipx
python -m pipx ensurepath

# Install ci_credentials
pipx install --editable --force .
```

**Comment by aaronsteers** [comment]
I spent some time in the Pyenv docs... Running `pyenv global 3.10 3.9 3.11` will ensure `3.10` is the default when you invoke python but it will also enable to use the shims `python3.9` and `python3.11` when a different version is needed. 

Ignoring 3.11 for now, the instructions could be...

Ensure `python3.9` and `python3.10` are both available as shims:

```
pyenv install 3.9
pyenv install 3.10
pyenv global 3.9 3.10
```

Installing pipx:
```
python3.10 -m pip install pipx
python3.10 -m pipx ensurepath

pipx install --editable --python=python3.10 airbyte-ci/connectors/pipelines
```

**Comment by aaronsteers** [comment]
Also, in my debugging, it appeared to me that the `--editable` flag did not carry over to the `connector_ops` sibling package, even though it is defined as a `path` reference with `develop = true` in `pyproject.toml`.

This is why I've started advising using the `pipx reinstall` option to refresh. Since that reliably gets the latest code installed, without being affected by lack of a version bump on related packages.

**Comment by bnchrch** [comment]
Ok so minor update here.

I dont think pyenv behaves like we would expect, and seems to be broken ( maybe just on my machine)

**The case of two python versions one file**
If I had a `.python-version` like so
```
3.10
3.9
3.11
```

and ran pyenv install

I would **expect**
1. 3.10, 3.9 and 3.11 would be installed
2. python3.10, python3.9, and python3.11 would be commands pointed at their repective python executables
3. python would point at python3.10

What **actually** occurs is
1. 3.10, 3.9 and 3.11 are installed ✅ 
2. Only python3.10 works as a command (the other two are empty shims) ⚠️ 
3. python would point at python3.10 ✅ 

Conclusion:
We should only use one python version per file. Others are relatively useless.

**What about pyenv global syste**
After looking at the [docs](https://github.com/pyenv/pyenv#understanding-python-version-selection)

I saw this
> NOTE: You can activate multiple versions at the same time, including multiple versions of Python2 or Python3 simultaneously. This allows for parallel usage of Python2 and Python3, and is required with tools like tox. For example, to instruct Pyenv to first use your system Python and Python3 (which are e.g. 2.7.9 and 3.4.2) but also have Python 3.3.6, 3.2.1, and 2.5.2 available, you first pyenv install the missing versions, then set pyenv global system 3.3.6 3.2.1 2.5.2. Then you'll be able to invoke any of those versions with an appropriate pythonX or pythonX.Y name. You can also specify multiple versions in a .python-version file by hand, separated by newlines. Lines starting with a # are ignored.

which indicated that by running `pyenv global system 3.9 3.10 3.11` I could fix the failure from the previous case
> Only python3.10 works as a command (the other two are empty shims) ⚠️ 

However that seemed to go no where. python3.9 and python3.11 were still empty shims.

image

**New propose plan**
1. Update pipx installs to occur in the folder of the tool itself e.g `pipx install .`
2. Readd a .python-version file targeting 3.10 in the root
    1.  @aaronsteers If im reading the code right. We dont really need python3.9, the exception being the octavia-cli? which is rarely built and has its own `.python-version` file.

**Comment by bnchrch** [comment]
Ok so minor update here.

I dont think pyenv behaves like we would expect, and seems to be broken ( maybe just on my machine)

**The case of two python versions one file**
If I had a `.python-version` like so
```
3.10
3.9
3.11
```

and ran pyenv install

I would **expect**
1. 3.10, 3.9 and 3.11 would be installed
2. python3.10, python3.9, and python3.11 would be commands pointed at their repective python executables
3. python would point at python3.10

What **actually** occurs is
1. 3.10, 3.9 and 3.11 are installed ✅ 
2. Only python3.10 works as a command (the other two are empty shims) ⚠️ 
3. python would point at python3.10 ✅ 

Conclusion:
We should only use one python version per file. Others are relatively useless.

**What about pyenv global**
After looking at the [docs](https://github.com/pyenv/pyenv#understanding-python-version-selection)

I saw this
> NOTE: You can activate multiple versions at the same time, including multiple versions of Python2 or Python3 simultaneously. This allows for parallel usage of Python2 and Python3, and is required with tools like tox. For example, to instruct Pyenv to first use your system Python and Python3 (which are e.g. 2.7.9 and 3.4.2) but also have Python 3.3.6, 3.2.1, and 2.5.2 available, you first pyenv install the missing versions, then set pyenv global system 3.3.6 3.2.1 2.5.2. Then you'll be able to invoke any of those versions with an appropriate pythonX or pythonX.Y name. You can also specify multiple versions in a .python-version file by hand, separated by newlines. Lines starting with a # are ignored.

which indicated that by running `pyenv global system 3.9 3.10 3.11` I could fix the failure from the previous case
> Only python3.10 works as a command (the other two are empty shims) ⚠️ 

However that seemed to go no where. python3.9 and python3.11 were still empty shims.

**Example of it not working like I would hope**
image

@aaronsteers Just to dummy check myself If you run these same commands do you end up in the same state?

**New propose plan**
1. Update pipx installs to occur in the folder of the tool itself e.g `pipx install .`
2. Readd a .python-version file targeting 3.10 in the root
    1.  @aaronsteers If im reading the code right. We dont really need python3.9, the exception being the octavia-cli? which is rarely built and has its own `.python-version` file.
3. If we can get python install / python global working right, then add the secodary version to the top level .python-version file

File: airbyte-ci/connectors/pipelines/pyproject.toml [hunk 2]
```diff
@@ -19,7 +19,7 @@ semver = "^3.0.1"
 airbyte-protocol-models = "*"
 tabulate = "^0.8.9"
 jinja2 = "^3.0.2"
-requests = "^2.28.2"
+requests = "2.28.2"
```
Comment (aaronsteers) [inline airbyte-ci/connectors/pipelines/pyproject.toml:22]: For our future ref: the logged issues relating to the conflict with `pydocker` recommend `<2.29.0`

**Review by pedroslopez — APPROVED** [review]

**Comment by octavia-approvington** [comment]
Lets merge it <br/> sheep thumbs up

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**